### ISSUE TITLE
Make nokogiri compatible to ruby 2.5 on Windows

### ISSUE THREAD

**Issue by SimonHoenscheid** [sparklemotion/nokogiri#1709]
## If you're having trouble installing Nokogiri ...

**Have you tried following [the installation tutorial][tutorial]?**

yes

**What is the output of `gem install`?**

```
PS C:\Users\replaced> gem install nokogiri -v 1.8.1
ERROR:  Error installing nokogiri:
The last version of nokogiri (= 1.8.1) to support your Ruby & RubyGems was 1.8.1. Try installing it with `gem install nokogiri -v 1.8.1`
 nokogiri requires Ruby version < 2.5, >= 2.2. The current ruby version is 2.5.0.
```

**What are the contents of the `mkmf.log` file?**

Didnt get so far

**What operating system are you using?**

Windows 10 x64

**Comment by francesco-loreti** [comment]
Same OS and same problem...

**Comment by fredzy07** [comment]
Same here, and I really would like it if downgrading is not my only option

**Comment by McRaeAlex** [comment]
I am currently having the same issue

**Comment by richpeck** [comment]
Same problem

**Comment by Shekharrajak** [comment]
Windows 7 x64 same error msg.

```
>gem install nokogiri -v 1.8.1
ERROR:  Error installing nokogiri:
        The last version of nokogiri (= 1.8.1) to support your Ruby & RubyGems w
as 1.8.1. Try installing it with `gem install nokogiri -v 1.8.1`
        nokogiri requires Ruby version < 2.5, >= 2.2. The current ruby version i
s 2.5.0.

```

**Comment by fera2k** [comment]
The fix is already merged, although a new release is pending. See: 
https://github.com/sparklemotion/nokogiri/pull/1704

Until then, it's better to stick with Ruby 2.4.2 on Windows.

The Nokogiri team removed the GEMSPEC from sources, to disallow people building from master branch, since it is unstable and probably would contain bugs. You can read more about it here:
https://github.com/sparklemotion/nokogiri/blob/master/Y_U_NO_GEMSPEC.md

**Comment by leehanslim** [comment]
Any word on support for 2.5.0 on OS X?

**Comment by tchikuba** [comment]
Same problem.

**Comment by francesco-loreti** [comment]
any news?

**Comment by SimonHoenscheid** [comment]
I am wondering how this happens. Its kind of a tradition ruby makes a point release around christmas every year. the dev version was around for weeks. Ignoring Windows blames the users who are bound to this OS for what ever reason

**Comment by flavorjones** [comment]
@SimonHoenscheid nokogiri-core is always welcoming of contributors who would like to help us support Windows. We've asked for help repeatedly over the years, as we've all got jobs that distract us from supporting open source software, despite our best efforts and intentions.

I'd like to ask that you pay attention to the tone of your message above. "Ignoring Windows" is clearly an untrue statement, given the number of hours I've personally spent, the number of hours @larskanis has spent, the amount of time we've spent providing [automated testing tooling][1] for [Nokogiri on Windows][2], etc., etc.

Windows support in Nokogiri is first class. We provide precompiled DLLs for users who want a fast, simple installation; and we provide support for users who want to compile Nokogiri and libxml using DevKit. Saying that we're ignoring Windows users is absurd and a little insulting.

Currently I'm blocking a release that supports Ruby 2.5 on windows on getting Ruby 2.5 supported in our test pipelines. I accept some fault for not being more transparent about this as the blocking factor. But now I'm trying to make amends by communicating exactly what's going on behind the scenes to properly support you and the platform that you use. Not ignoring you or blaming you, as you claim.

I'm going to lock this issue because I really don't want to do a whole back-and-forth. We'll release support for Ruby 2.5 on Windows as soon as we can, just as we always do, and we appreciate your patience and empathy as we work.

 [1]: https://github.com/flavorjones/windows-ruby-dev-tools-release
 [2]: https://ci.nokogiri.org/teams/nokogiri-core/pipelines/nokogiri/jobs/win-ruby-2.4-devkit/builds/10

**Comment by flavorjones** [comment]
Remaining work to be done on our CI pipeline at https://ci.nokogiri.org/

* [ ] [upgrade concourse to 3.8.0](https://concourse.ci/downloads.html#v380)
* [ ] test Ruby 2.5 support just added to [windows-ruby-dev-tools-release](https://github.com/flavorjones/windows-ruby-dev-tools-release/tree/flavorjones-add-ruby-25) and cut a release
* [ ] add a Ruby 2.5 test job to the pipeline
* [ ] cut a Nokogiri release that adds Ruby 2.5

This is probably a good time to remind watchers that Ruby 2.2 is approaching EOL on  2018-03-31, and so the first release after that will likely remove Ruby 2.2 binaries from the fat gem.

**Comment by flavorjones** [comment]
Windows Ruby 2.5 build is up and running:

> https://ci.nokogiri.org/teams/nokogiri-core/pipelines/nokogiri/jobs/win-ruby-2.5-devkit/builds/1

**Comment by flavorjones** [comment]
Note that commit bf94cf5 was added to master to make windows tests less brittle.

**Comment by flavorjones** [comment]
Nokogiri 1.8.2 has shipped with Ruby 2.5 support in the "fat binary" windows gems. Thanks everyone for your patience.
